### DISCUSSION TITLE
词典笔P5获取ADB密码过程中遇到难题，密码不是明文保存，而是sha256

### DISCUSSION THREAD

**Discussion by XSMiracle** [PenUniverse/PenMods#187] [你问我答]
我在词典笔P5获取ADB密码的过程中，用抓包改请求的方式确实拿到了img文件，但是打开之后发现密码已经不是被明文保存了，而是换成了sha256验证的方式
![IMG_6371](https://github.com/PenUniverse/PenMods-release/assets/111876476/a354f49e-802b-4f72-903a-5420a42e14f2)
我的这个sha256值是bdb53c0b78a541e5203e1a009bc0f2723989b2bf478493ee5fd4e2bb03e19d1e，请问各位大佬有没有什么好办法（）

**Comment by EXE-autumnwind** [comment]
关机后按下home+电源键+笔头降级试试看

**Reply by XSMiracle** [reply]
可以详细说说吗

**Reply by EXE-autumnwind** [reply]
我也不太清楚，只是让你回退到出产版本试试，这个降级方法在ydpen2.0试过，p5不太清楚

**Comment by dotinx** [comment]
欸 你解的哪个更新包？我这边 3.4.0 3.4.2 3.4.4 的增量更新包中均没有此文件……
如果真的很需要 ADB 可以试试劫持更新包，往 init.d 里面扔个脚本（注意得在 mount tmp 之后）自动 `touch /tmp/.adb_auth_verified`

**Reply by XSMiracle** [reply]
3.4.0，通过抓包然后改post请求体里面version参数再用post测试网站请求拿到的下载url，然后用imgRePacker解包的img，然后进入目录有一个rootfs.img，直接打开在usr/bin目录下找到的adb_auth.sh文件

**Reply by dotinx** [reply]
那我懷疑有道更改了他們的分發策略以分發了一個新的3.4.0版本。
[我在23年四月提取的更新包](https://pan.inx.wf/onedrive/Youdao%20P5/Update%20Image/3.4.0)
![image](https://github.com/PenUniverse/PenMods-release/assets/115958651/560002e8-0054-40a4-8474-62232f368a0c)
該包的 Version 信息爲
```
Firmware:
DictPen_APP:
Build_Time:二 3月 14 20:51:20 HKT 2023
Version: 3.4.0
```

**Reply by dotinx** [reply]
https://t.me/PenUniverse/4977

https://dotinx-my.sharepoint.com/:f:/g/personal/inx_inx_wf/EmoxbcQFLQJFiQJRFjXCInEB0dJkmkEoOS_EQSoxZQyC5A?e=RfFNFJ

這是我手頭所有的鏡像

https://dotinx-my.sharepoint.com/:u:/g/personal/inx_inx_wf/EYDw2P3OHRZCsLWU7_YnF2IB9Iel4KhFbCGyaeArqeReOg?e=I6JUU3

這是我手頭的 3.4.0 ，解包發現無 adb_auth.sh

**Reply by wyxdlz54188** [reply]
有没有A6 Pro的破解方法，已经改为md5加密

**Reply by LuorixDev** [reply]
> 有没有A6 Pro的破解方法，已经改为md5加密

+1 我也需要

**Reply by Jerry12228** [reply]
https://github.com/orgs/PenUniverse/discussions/250
